rust/worker/src/lib.rs

```diff
@@ -145,11 +146,11 @@ pub async fn compaction_service_entrypoint() {
         // TODO: add more signal handling
         _ = sigterm.recv() => {
             memberlist_handle.stop();
-            memberlist_handle.join().await;
+            let _ = memberlist_handle.join().await;
             dispatcher_handle.stop();
-            dispatcher_handle.join().await;
+            let _ = dispatcher_handle.join().await;
             compaction_manager_handle.stop();
-            compaction_manager_handle.join().await;
+            let _ = compaction_manager_handle.join().await;
```

Comment:
would unwrapping be more appropriate here (and above)?

Reply:
I kept the "drop" behavior.  If you think it'll be safe (e.g., should never fail), I'll add a comment/unwrap.

Reply:
It's not safe. This is the server shutdown. The join can fail. Given that, a failed join probably warrants crashing the server.